framework/include/outputs/OutputWarehouse.h

```diff
@@ -189,6 +189,43 @@ class OutputWarehouse
    */
   void bufferConsoleOutputsBeforeConstruction(bool buffer) { _buffer_action_console_outputs = buffer; }
 
+public:
+  /**
+   * Calls the initialSetup function for each of the output objects
+   * @see FEProblemBase::initialSetup()
+   */
+  void initialSetup();
```

Comment:
Not necessarily.  There is nothing inherently wrong with `friend`.  It all depends on the context.  Making things `public` means that anyone is free to go at that function... `friend` can be used to allow internal objects to work together without exposing functionality to the whole world.

In our case with MOOSE I often use `friend` so that I can allow MOOSE internal classes more access to each other while shielding our users (or shielding ourselves from our users!) from using that capability.